src/com/palmergames/util/Trie.java

```diff
@@ -144,17 +132,21 @@ public boolean removeKey(String key) {
 		nodes.put(root, ""); // Start with the root node
 
 		for (int i = 0; i < key.length(); i++) {
-			int finalI = i;
 			Map<TrieNode, String> newNodes = new HashMap<>(); // An updated version of nodes, will not contain the old values
+			char index = Character.toLowerCase(key.charAt(i));
+
 			for (Map.Entry<TrieNode, String> entry : nodes.entrySet()) { // Loop through the old nodes
-				List<TrieNode> list = entry.getKey().children.stream()
-					.filter(e -> Character.toLowerCase(e.character) == Character.toLowerCase(key.charAt(finalI))).collect(Collectors.toList()); // Find matches for lower and upper case
-
-				for (TrieNode listNode : list) { // Loop through all matches, could be a list of 1 or 2 - for example if we looked for "e" and the node had children "e" and "E" the list would have 2 values
-					newNodes.put(listNode, entry.getValue()+listNode.character); // entry.getValue is the old key for the node, for example "bana" as entry.getValue() and "n" as listNode.character resulting in "banan" for listNode
-					if (i == key.length() - 1) { // Check if this is the last character of the key, indicating a word ending. From here we need to find all the possible children
-						for (String string : getChildrenStrings(listNode, new ArrayList<>())) { // Recursively find all children
-							strings.add(newNodes.get(listNode) + string); // Add the key to the front of each child string
+
+				for (TrieNode node : entry.getKey().children) {
+
+					if (Character.toLowerCase(node.character) == index) {
+						newNodes.put(node, entry.getValue()+node.character); // entry.getValue is the old key for the node, for example "bana" as entry.getValue() and "n" as listNode.character resulting in "banan" for listNode
```

Comment:
You could probably store `entry.getValue() + node.character` in a variable to prevent having to call `newNodes.get()` in the code below.

Reply:
Changed, thanks for the catch